src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -218,12 +218,17 @@ public override string ToString()
         }
 
         // internal test hook
-        private static bool s_forceNullEntryPoint;
+        private static Assembly? s_overriddenEntryAssembly;
+
+        public static void SetEntryAssembly(Assembly newAssembly)
```

Comment:
The approved argument name is `assembly`.

Also, I think this should allow passing in `null` to clear the entry assembly.

Reply:
Is there a reason or use case we would want to clear the entry assembly entirely? I see it from the perspective that a dotnet app will always have an entry assembly, otherwise how is it executing, so I think clearing the entry assembly would just be inconsistent with the program's state.

Reply:
It's null for various hosting scenarios or e.g. native libraries: https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/libraries

Reply:
> I see it from the perspective that a dotnet app will always have an entry assembly

The entry assembly can be null in hosted scenarios when you start the runtime via hosting APIs (https://learn.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting) and not by launching an app.

Notice that the existing `Assembly.GetEntryAssembly` returns `Assembly?`.

> Is there a reason or use case we would want to clear the entry assembly entirely? 

One use case are tests that exercise the hosting scenario, without actually hosting the runtime. You can see we have number of those in this repo.